ext/spl/tests/gh14687.phpt

```diff
@@ -0,0 +1,44 @@
+--TEST--
+GH-14687 segfault on debugging SplObjectStorage instance after __destruct.
+--CREDITS--
+YuanchengJiang
+--EXTENSIONS--
+phar
+--INI--
+phar.require_hash=0
+phar.readonly=0
+--FILE--
+<?php
+$fname = __DIR__ . '/' . basename(__FILE__, '.php') . '.phar.zip';
+$phar = new Phar($fname);
+class HasDestructor {
+  public function __destruct() {
+    var_dump($GLOBALS['s']);
+  }
+}
+$s = new SplObjectStorage();
+$s[$phar] = new HasDestructor();
+register_shutdown_function(function() {
+	global $s;
+});
```

Comment:
Please close the test with `?>`, this is necessary so we can run the test directly.